test/IDE/complete_expr_tuple.swift

```diff
@@ -27,27 +27,45 @@ func testTupleNoDot1() {
   var t = (1, 2.0)
   t#^TUPLE_NO_DOT_1^#
 }
-// TUPLE_NO_DOT_1: Begin completions, 2 items
+// TUPLE_NO_DOT_1: Begin completions, 8 items
 // TUPLE_NO_DOT_1-NEXT: Pattern/CurrNominal: .0[#Int#]{{; name=.+$}}
 // TUPLE_NO_DOT_1-NEXT: Pattern/CurrNominal: .1[#Double#]{{; name=.+$}}
+// TUPLE_NO_DOT_1-NEXT: Decl[InfixOperatorFunction]/OtherModule[Swift]: == {#(Int, Double)#}[#Bool#]{{; name=.+$}}
+// TUPLE_NO_DOT_1-NEXT: Decl[InfixOperatorFunction]/OtherModule[Swift]: <= {#(Int, Double)#}[#Bool#]{{; name=.+$}}
+// TUPLE_NO_DOT_1-NEXT: Decl[InfixOperatorFunction]/OtherModule[Swift]: >= {#(Int, Double)#}[#Bool#]{{; name=.+$}}
+// TUPLE_NO_DOT_1-NEXT: Decl[InfixOperatorFunction]/OtherModule[Swift]: < {#(Int, Double)#}[#Bool#]{{; name=.+$}}
+// TUPLE_NO_DOT_1-NEXT: Decl[InfixOperatorFunction]/OtherModule[Swift]: != {#(Int, Double)#}[#Bool#]{{; name=.+$}}
+// TUPLE_NO_DOT_1-NEXT: Decl[InfixOperatorFunction]/OtherModule[Swift]: > {#(Int, Double)#}[#Bool#]{{; name=.+$}}
```

Comment:
How are code completions sorted? I tried looking through the code but it's not immediately obvious. I did find code that appears to sort them based on some definition of a name (which I'd assume is like `"=="` but I don't actually know), but the output here doesn't appear to be sorted in any way I can figure out. If it was sorted by name I'd expect to see the ordering be `!=`, `<`, `<=`, `==`, `>`, `>=`, and it's not. Heck, `!=` should show up before `.0` then (and in fact when tab-completing in the integrated REPL it does sort it like that). So the ordering here must be based on something else, and I'm worried that this "something else" might not be stable.

I'm also not sure if we really want to match the full Decl[...]/OtherModule[..] prefix, but I'm not as concerned about that.

Ideally I'd actually switch to using `-DAG` patterns to make it unordered, but that loses the property of ensuring there's no lines before the `End completions` that don't match one of the expected completions.